src/OrbitGl/TimerTrack.cpp

```diff
@@ -82,6 +82,147 @@ WorldXInfo ToWorldX(double start_us, double end_us, double inv_time_window, floa
+      if (current_timer_info.end() > next_timer_info.start() &&
+          current_timer_info.end() <= next_timer_info.end() &&
+          next_timer_info.type() == current_timer_info.type()) {
+        end_or_next_start_us = time_graph_->GetUsFromTick(next_timer_info.start());
+      }
+    }
+  }
+
+  double elapsed_us = end_us - start_us;
+
+  // Draw the Timer's text if it is not collapsed.
+  if (!is_collapsed) {
+    // For overlaps, let us make the textbox just a bit wider:
+    double left_overlap_width_us = start_or_prev_end_us - start_us;
+    double text_x_start_us = start_or_prev_end_us - (.25 * left_overlap_width_us);
+    double right_overlap_width_us = end_us - end_or_next_start_us;
+    double text_x_end_us = end_or_next_start_us + (.25 * right_overlap_width_us);
+
+    bool is_visible_width =
+        (text_x_end_us - text_x_start_us) * inv_time_window * canvas->GetWidth() > 1;
+    WorldXInfo world_x_info =
+        ToWorldX(text_x_start_us, text_x_end_us, inv_time_window, world_start_x, world_width);
+
+    if (is_visible_width) {
+      Vec2 pos(world_x_info.world_x_start, world_timer_y);
+      Vec2 size(world_x_info.world_x_width, GetTextBoxHeight(current_timer_info));
+      current_text_box->SetPos(pos);
+      current_text_box->SetSize(size);
+
+      SetTimesliceText(current_timer_info, elapsed_us, world_start_x, z_offset, current_text_box);
+    }
+  }
+
+  uint64_t function_id = current_timer_info.function_id();
+
+  bool is_selected = current_text_box == selected_textbox;
+  bool is_highlighted = !is_selected && function_id != orbit_grpc_protos::kInvalidFunctionId &&
+                        function_id == highlighted_function_id;
+
+  static const Color kHighlightColor(100, 181, 246, 255);
+  Color color = is_highlighted ? kHighlightColor : GetTimerColor(current_timer_info, is_selected);
+
+  bool is_visible_width = elapsed_us * inv_time_window * canvas->GetWidth() > 1;
+
+  if (is_visible_width) {
+    WorldXInfo world_x_info_left_overlap =
+        ToWorldX(start_us, start_or_prev_end_us, inv_time_window, world_start_x, world_width);
+
+    WorldXInfo world_x_info_right_overlap =
+        ToWorldX(end_or_next_start_us, end_us, inv_time_window, world_start_x, world_width);
+
+    Vec3 top_left(world_x_info_left_overlap.world_x_start, world_timer_y + box_height, z);
+    Vec3 bottom_left(
+        world_x_info_left_overlap.world_x_start + world_x_info_left_overlap.world_x_width,
+        world_timer_y, z);
+    Vec3 top_right(world_x_info_right_overlap.world_x_start, world_timer_y + box_height, z);
+    Vec3 bottom_right(
+        world_x_info_right_overlap.world_x_start + world_x_info_right_overlap.world_x_width,
+        world_timer_y, z);
+    batcher->AddShadedTrapezium(
+        top_left, bottom_left, bottom_right, top_right, color,
+        std::make_unique<PickingUserData>(current_text_box,
+                                          [&](PickingId id) { return this->GetBoxTooltip(id); }));
+
+  } else {
```

Comment:
pers pref:
```suggestion
  return
  } 
```

Reply:
The return is inside the lambda, so I guess I can't do the early return here.